cinema/views.py

```diff
@@ -21,24 +30,32 @@
 )
 
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(CreateModelMixin, GenericViewSet, ListModelMixin):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    permission_classes = [user.permissions.IsAdminOrIfAuthenticatedReadOnly]
+    authentication_classes = (authentication.TokenAuthentication,)
```

Comment:
instead of duplicating code, you can set DEFAULT_PERMISSION_CLASSES and DEFAULT_AUTHENTICATION_CLASSES in REST_FRAMEWORK constant in settings.py and override these attributes where needed